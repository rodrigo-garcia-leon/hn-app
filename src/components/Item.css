.Item h1 {
  color: #444444;
  font-size: 18px;
  font-weight: 400;
  text-decoration: none;
}

.Item h2 {
  color: #bbbbbb;
  font-size: 14px;
}
