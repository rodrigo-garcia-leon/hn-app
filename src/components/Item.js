import React from "react";
import { formatTime } from "../shared/utils";
import "./Item.css";

function Item({ item }) {
  if (!item) {
    return null;
  }

  const { by, score, time, title } = item;

  return (
    <section className="Item">
      <h1>{title}</h1>
      <h2>
        {score} points by {by} {formatTime(time)} ago
      </h2>
    </section>
  );
}

export default Item;
