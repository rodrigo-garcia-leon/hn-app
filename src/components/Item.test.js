import React from "react";
import { render } from "@testing-library/react";
import Item from "./Item";
import { TEST_STORY_22089546 } from "hn-api";

jest.mock("../shared/utils.js");

describe.only("Item", () => {
  test("empty", () => {
    const element = render(<Item />);

    expect(element.container).toMatchSnapshot();
  });

  test("ok", () => {
    const element = render(<Item item={TEST_STORY_22089546} />);

    expect(element.container).toMatchSnapshot();
  });
});
