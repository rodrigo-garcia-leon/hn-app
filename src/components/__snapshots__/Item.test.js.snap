// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item empty 1`] = `<div />`;

exports[`Item ok 1`] = `
<div>
  <section
    class="Item"
  >
    <h1>
      GCC: C++ coroutines â€“ Initial implementation pushed to master
    </h1>
    <h2>
      67
       points by 
      matt_d
       
      8 hours
       ago
    </h2>
  </section>
</div>
`;
